File: src/main/webapp/js/modulosPagos/TipoPago.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function TipoPago(id, concepto, conceptoPara, esInhabilitadora, numPagosActivos, tipoGeneracion,
    estado, estadosColegiados, precioActual, precioEspera, preciosHistoricos){
    this.id = id;
    this.concepto = concepto;
    this.conceptoPara = conceptoPara;
    this.esInhabilitadora = esInhabilitadora;
    this.numPagosActivos = numPagosActivos;
    this.tipoGeneracion = tipoGeneracion;
    this.estado = estado;
    this.estadosColegiados = estadosColegiados;
    this.precioActual = precioActual;
    this.precioEspera = precioEspera;
    this.preciosHistoricos = preciosHistoricos;
    this.indError = "";
    this.msjError = "";
    this.listarTipoPago = listarTipoPago;
    this.mantTipoPago = mantTipoPago;
}

function ListTipoPago(listTipoPago, indError, msjError){
    this.listTipoPago = listTipoPago;
    this.indError = indError;
    this.msjError = msjError;
}


function listarTipoPago(tipoOperacion){
    var listaTipoPagoRep;
    var tipoPago ={
        "id": this.id,
        "conceptoPara" : this.conceptoPara,
        "estado" : this.estado
    };
    listaTipoPagoRep = new ListTipoPago();
    $.ajax({
        url : "../pagos/listarTipoPago.action",
        type : "POST",
        contentType : 'application/json; charset=utf-8',
        dataType : 'json',
        async: false,
        mimeType: 'application/json',
        data : JSON.stringify({tipoOperacion: tipoOperacion, tipoPago : tipoPago}),
        success : function(data) {
            listaTipoPagoRep.listTipoPago = data.listTipoPago;
            listaTipoPagoRep.indError = data.indError;
            listaTipoPagoRep.msjError = data.msjError;
        }
    });
    return listaTipoPagoRep;
}

function mantTipoPago(tipoOperacion){
    var tipoPagoRep;
    var tipoPago ={
        "id" : this.id,
        "concepto" : this.concepto,
        "conceptoPara" : this.conceptoPara,
        "esInhabilitadora" : this.esInhabilitadora,
        "numPagosActivos" : this.numPagosActivos,
        "tipoGeneracion" : this.tipoGeneracion,
        "estado" : this.estado,
        "estadosColegiados" : this.estadosColegiados,
        "precioActual" : this.precioActual
    };
    tipoPagoRep = new TipoPago();
    $.ajax({
        url : "../pagos/mantTipoPago.action",
        type : "POST",
        contentType : 'application/json; charset=utf-8',
        dataType : 'json',
        async: false,
        mimeType: 'application/json',
        data : JSON.stringify({tipoOperacion: tipoOperacion, tipoPago : tipoPago}),
        success : function(data) {
            tipoPagoRep.indError = data.indError;
            tipoPagoRep.msjError = data.msjError;
        }
    });
    return tipoPagoRep;
}

File: src/main/java/com/whnm/sicqfdp/beans/TipoPago.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.whnm.sicqfdp.beans;

import java.util.List;

/**
 *
 * @author whnm
 */
public class TipoPago {
    private Long id;
    private String concepto;
    private String conceptoPara;
    private Integer esInhabilitadora;
    private Integer numPagosActivos;
    private Integer tipoGeneracion;
    private Integer estado;
    private String estadosColegiados;
    private Integer annadido;
    private Integer tipoCantidad;
    private Precio precioActual;
    private Precio precioEspera;
    private List<Precio> preciosHistoricos;
    private Integer indTablaPrecio;
    private Integer indTmpPrecio;
    private Integer indError;
    private String msjError;

    public Integer getIndTablaPrecio() {
        return indTablaPrecio;
    }

    public void setIndTablaPrecio(Integer indTablaPrecio) {
        this.indTablaPrecio = indTablaPrecio;
    }

    public Integer getIndTmpPrecio() {
        return indTmpPrecio;
    }

    public void setIndTmpPrecio(Integer indTmpPrecio) {
        this.indTmpPrecio = indTmpPrecio;
    }

    public TipoPago() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getConcepto() {
        return concepto;
    }

    public void setConcepto(String concepto) {
        this.concepto = concepto;
    }

    public String getConceptoPara() {
        return conceptoPara;
    }

    public void setConceptoPara(String conceptoPara) {
        this.conceptoPara = conceptoPara;
    }

    public Integer getEsInhabilitadora() {
        return esInhabilitadora;
    }

    public void setEsInhabilitadora(Integer esInhabilitadora) {
        this.esInhabilitadora = esInhabilitadora;
    }

    public Integer getNumPagosActivos() {
        return numPagosActivos;
    }

    public void setNumPagosActivos(Integer numPagosActivos) {
        this.numPagosActivos = numPagosActivos;
    }

    public Integer getTipoGeneracion() {
        return tipoGeneracion;
    }

    public void setTipoGeneracion(Integer tipoGeneracion) {
        this.tipoGeneracion = tipoGeneracion;
    }

    public Integer getEstado() {
        return estado;
    }

    public void setEstado(Integer estado) {
        this.estado = estado;
    }

    public String getEstadosColegiados() {
        return estadosColegiados;
    }

    public void setEstadosColegiados(String estadosColegiados) {
        this.estadosColegiados = estadosColegiados;
    }

    public Integer getAnnadido() {
        return annadido;
    }

    public void setAnnadido(Integer annadido) {
        this.annadido = annadido;
    }
    
    public Precio getPrecioActual() {
        return precioActual;
    }

    public void setPrecioActual(Precio precioActual) {
        this.precioActual = precioActual;
    }

    public Precio getPrecioEspera() {
        return precioEspera;
    }

    public void setPrecioEspera(Precio precioEspera) {
        this.precioEspera = precioEspera;
    }

    public List<Precio> getPreciosHistoricos() {
        return preciosHistoricos;
    }

    public void setPreciosHistoricos(List<Precio> preciosHistoricos) {
        this.preciosHistoricos = preciosHistoricos;
    }

    public Integer getTipoCantidad() {
        return tipoCantidad;
    }

    public void setTipoCantidad(Integer tipoCantidad) {
        this.tipoCantidad = tipoCantidad;
    }

    public Integer getIndError() {
        return indError;
    }

    public void setIndError(Integer indError) {
        this.indError = indError;
    }

    public String getMsjError() {
        return msjError;
    }

    public void setMsjError(String msjError) {
        this.msjError = msjError;
    }
    
    
}
